src/actions/receive-profile.js

```diff
@@ -156,15 +162,50 @@ export function finalizeProfileView(
     // been seen before. If it's non-null, then there is profile view information
     // encoded into the URL.
     const selectedThreadIndex = getSelectedThreadIndexOrNull(getState());
+    const pages = profile.pages;
+    if (!timelineTrackOrganization) {
+      // Most likely we'll need to load the timeline track organization, as requested
+      // by the URL, but tests can pass in a value.
+      timelineTrackOrganization = getTimelineTrackOrganization(getState());
+    }
 
-    if (selectedThreadIndex !== null && getShowTabOnly(getState()) !== null) {
-      // The url state says this is an active tab view. We should compute and
-      // initialize the state relevant to that state.
-      dispatch(finalizeActiveTabProfileView(profile, selectedThreadIndex));
-    } else {
-      // The url state says this is a full view. We should compute and initialize
-      // the state relevant to that state.
-      dispatch(finalizeFullProfileView(profile, selectedThreadIndex));
+    switch (timelineTrackOrganization.type) {
+      case 'full':
+        // The url state says this is a full view. We should compute and initialize
+        // the state relevant to that state.
+        dispatch(finalizeFullProfileView(profile, selectedThreadIndex));
+        break;
+      case 'active-tab':
+        if (selectedThreadIndex === null) {
+          // Switch back over to the full view.
+          timelineTrackOrganization = { type: 'full' };
+        } else {
+          // The url state says this is an active tab view. We should compute and
+          // initialize the state relevant to that state.
+          dispatch(
+            finalizeActiveTabProfileView(
+              profile,
+              selectedThreadIndex,
+              timelineTrackOrganization.browsingContextID
+            )
+          );
+        }
+        break;
+      case 'origins': {
+        if (pages) {
```

Comment:
Oh I like that check. We should also add this for active tab. (I think we handle this somewhere else inside, but still would be good to return early.)

Reply:
The active tab also requires a the browsing context, so it's a bit different. We should validate that piece of information, and switch back to origins if it doesn't check out too.